rpc/json/service.go

```diff
@@ -103,13 +134,10 @@ func (s *service) Subscribe(req *http.Request, args *subscribeArgs, wsConn *wsCo
 
 	s.logger.Debug("subscribe to query", "remote", addr, "query", args.Query)
 
-	// TODO(tzdybal): extract consts or configs
-	const SubscribeTimeout = 5 * time.Second
-	const subBufferSize = 100
-	ctx, cancel := context.WithTimeout(req.Context(), SubscribeTimeout)
+	ctx, cancel := context.WithTimeout(req.Context(), s.subscribeTimeout)
 	defer cancel()
 
-	sub, err := s.client.EventBus.Subscribe(ctx, addr, q, subBufferSize)
+	sub, err := s.client.EventBus.Subscribe(ctx, addr, q, s.subscribeBufferSize)
```

Comment:
* why is `Subscribe` called directly on the `EventBus` and not `s.client.Subscribe`?
* why we adding timeout only on the `Subscribe` interact and not on all other queries?

Reply:
I wouldn't know, I was just replacing the magic numbers.

Reply:
now looking at the code, 
- `s.client.Subscribe` is not exactly the same as `s.client.EventBus.Subscribe`. The former has a different signature and does a bit more (publish events and whatnot)
- you suggest we add it to `Unsubscribe` and `UnsubscribeAll` ?  Apart from them, only `TxSearch`, `BlockSearch`, `BroadcastTxCommit` and `BroadcastTxSync` actually use the passed context

Reply:
Makes sense.

Reply:
I just wish the implementation didn't rely so heavily on reflection